core/transaction_pool.go

```diff
@@ -71,7 +71,12 @@ type TxPool struct {
 	queue   map[common.Address]map[common.Hash]*types.Transaction
 }
 
-func NewTxPool(eventMux *event.TypeMux, currentStateFn stateFn, gasLimitFn func() *big.Int) *TxPool {
+func NewTxPool(eventMux *event.TypeMux, currentStateFn stateFn, gasLimitFn func() *big.Int) (*TxPool, error) {
+	currentState, err := currentStateFn()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Why is it unfortunate? It shouldn't change any behaviour of the pool, since if it happens it will simply refuse to start the pool.